[quant][be] Clean up prepare code

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #111828
* __->__ #111827

Summary:
att

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][be] Clean up prepare code

Summary:
att

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

andrewor14 commented on torch/ao/quantization/pt2e/prepare.py:
```diff
@@ -412,11 +399,8 @@ def _maybe_insert_input_and_output_observers_for_node(
-    if skip_inserting_output_observers:
+    output_is_a_tensor = isinstance(node.meta["val"], FakeTensor) if "val" in node.meta else False
+    if not output_is_a_tensor:
         return
```
This check is slightly hard to read. Why not just:
```
output_is_a_tensor = "val" in node.meta and isinstance(node.meta["eval"], FakeTensor)
```

jerryzh168 committed:
Update on "[quant][be] Clean up prepare code"


Summary:
att

Test Plan:
CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge
